framework/tlvf/src/src/BaseClass.cpp

```diff
@@ -44,24 +44,3 @@ bool BaseClass::buffPtrIncrementSafe(size_t length)
     m_buff_ptr__ += length;
     return true;
 }
-
-bool BaseClass::finalize()
-{
-    if (m_parse__) {
-        TLVF_LOG(ERROR) << "finalize() called but m_parse__ is set";
-        return false;
-    }
-    if (m_finalized) {
-        TLVF_LOG(DEBUG) << "finalize() called for already finalized class";
-        return true;
-    }
-    if (!isPostInitSucceeded()) {
-        TLVF_LOG(ERROR) << "post init check failed";
-        return false;
-    }
-    if (m_inner)
-        m_inner->finalize();
-    class_swap();
-    m_finalized = true;
-    return true;
-}
```

Comment:
I would prefer to keep this function in BaseClass, and add a private virtual finalize_impl function or something like that that is optionally generated by tlvf and that defaults to nothing. That keeps most of the code outside of tlvf which is nicer.

Reply:
Will be reviewed as part of #601